guillotina/auth/utils.py

```diff
@@ -44,11 +44,16 @@ def set_authenticated_user(user):
 async def find_user(token):
     if token.get("id") == ROOT_USER_ID:
         root = get_utility(IApplication, name="root")
-        return root.root_user
-    for identifier in app_settings["auth_user_identifiers"]:
-        user = await identifier().get_user(token)
-        if user is not None:
-            return user
+        user = root.root_user
+    else:
+        for identifier in app_settings["auth_user_identifiers"]:
+            user = await identifier().get_user(token)
+            if user is not None:
+                break
+    if user:
+        groups = get_utility(IGroups)
+        await groups.load_groups(user.groups)
+    return user
```

Comment:
@bloodbare As you said, we load the groups needed by the user

Reply:
@bloodbare and regarding your comment 
> if user groups are loaded on memory cache and if not load them.

This would cause having a stale permission/roles in the cache when the group is modified.
IMHO the cache can be reimplemented in a custom IGroups in the case of advanced users that need more performance.